vaultclient/src/test/java/com/scality/vaultclient/services/AccountServicesClientTest.java

```diff
@@ -89,15 +117,58 @@ public static void init() throws Exception{
+    }
 
-        basicAWSCredentials = new BasicAWSCredentials("accesskey", "secretkey");
-        accountServicesClient2 = new AccountServicesClient(amazonHttpClient, basicAWSCredentials);
+    private static void initListAccountMocks() {
+
+        listAccountAmazonHttpClient = mock(AmazonHttpClient.class);
+        listAccountsMockClient = new AccountServicesClient(listAccountAmazonHttpClient);
+
+        //Default Create Account mock response
+        when(listAccountAmazonHttpClient.execute(any(Request.class), any(), any(), any(),any()))
+                .thenAnswer(new Answer<Response>() {
+                    @Override
+                    public Response<ListAccountsResponseDTO> answer(InvocationOnMock invocation) {
+                        Request<ListAccountsRequestDTO> ogReq = invocation.getArgument(0);
+                        ListAccountsRequestDTO request = (ListAccountsRequestDTO) ogReq.getOriginalRequest();
+                        String marker = request.getMarker();
+                        int maxItems = request.getMaxItems();
+                        if(maxItems <= 0){
+                            maxItems = DEFAULT_LIST_ACCOUNTS_COUNT; // Test Accounts count
+                        }
+
+                        int i = 0;
+                        int markerVal = 0;
+                        if(StringUtils.isNotBlank(marker)){
+                            markerVal = Integer.parseInt(marker.substring(marker.length()-1));
+                            // extracting markerVal index at last character
+                        }
+
+                        List<AccountData> accounts = new ArrayList<>();
+                        // Generate Accounts with ids (markerVal + i) to maxItems count
+                        for(; i < maxItems; i++){
+                            AccountData data = new AccountData();
+                            data.setEmailAddress(DEFAULT_EMAIL_ADDR);
+                            data.setName(DEFAULT_ACCOUNT_NAME);
+                            data.setArn(DEFAULT_ARN_STR + DEFAULT_ACCOUNT_NAME +"/\"");
+                            data.setCreateDate(new Date());
+                            data.setId(DEFAULT_ACCOUNT_ID + (i + markerVal)); //setting ID with index
+                            data.setCanonicalId(DEFAULT_CANONICAL_ID);
+
+                            accounts.add(data);
+                        }
+
+                        ListAccountsResponseDTO response = new ListAccountsResponseDTO();
+                        response.setAccounts(accounts);
+                        return new Response<ListAccountsResponseDTO>(response,null);
+                    }
+                });
     }
 
     @Test
     public void testCreateAccount() throws Exception {
 
-        CreateAccountResponseDTO response = accountServicesClient.createAccount(createAccountRequestDTO).getAwsResponse();
+        CreateAccountResponseDTO response = createAccountMockClient.createAccount(createAccountRequestDTO).getAwsResponse();
```

Comment:
Shouldn't this just be, `accountMockClient.createAccount` as we are not creating it here, it's just a client.

Same in other places.